### ISSUE TITLE
cmd/link: Windows -H windowsgui stopped working in Go 1.10?

### ISSUE THREAD

**Issue by devorel** [golang/go#24232]
Please answer these questions before submitting your issue. Thanks!

### What did you do?

go build -ldflags "-H windowsgui"  -v -o %gopath%/bin/test.exe my\proj\test.io 

if i compile my program with version 1.9 , it's work good! 
but if i compile my program in the last version 1.10 , i get black window console

### What did you expect to see?
my window program without window console 

### What did you see instead?
my window program with window console

**Comment by alexbrainman** [comment]
@devorel I cannot reproduce this. I did this:
```
c:\>go version
go version go1.10 windows/amd64

c:\>go build -ldflags "-H windowsgui" -v -o %tmp%\a.exe github.com/alexbrainman/gowingui

c:\>
```
and then I double clicked on %TMP%\a.exe. I don't see any window console.

What did I do wrong?

Thank you.

Alex

**Comment by devorel** [comment]
You are use with shrot process.

If u use with server in the Background    you need hold  your process .
I dont need window console for that..



בתאריך 4 במרץ 2018 04:54,‏ "Alex Brainman" <notifications@github.com> כתב:

@devorel <https://github.com/devorel> I cannot reproduce this. I did this:

c:\>go version
go version go1.10 windows/amd64

c:\>go build -ldflags "-H windowsgui" -v -o %tmp%\a.exe
github.com/alexbrainman/gowingui

c:\>

and then I double clicked on %TMP%\a.exe. I don't see any window console.

What did I do wrong?

Thank you.

Alex

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/golang/go/issues/24232#issuecomment-370197648>, or mute
the thread

**Comment by alexbrainman** [comment]
@devorel I don't understand your explanation. Please, try again.

Thank you

Alex

**Comment by devorel** [comment]
look
this flug "windowsgui" need to cancel the window console

2018-03-04 8:11 GMT+02:00 Alex Brainman <notifications@github.com>:

**Comment by devorel** [comment]
look
this flug "windowsgui" need to cancel the window console

![hello1 10](https://user-images.githubusercontent.com/16879980/36943067-173ab4a6-1f8b-11e8-8286-090632fc8d9d.png)
![1 10v](https://user-images.githubusercontent.com/16879980/36943068-175e3d04-1f8b-11e8-8b7b-a82e1c55fb24.png)
![hello1 9](https://user-images.githubusercontent.com/16879980/36943069-177e1476-1f8b-11e8-8db0-b1c607007fa8.png)
![1 9v](https://user-images.githubusercontent.com/16879980/36943070-179e6a64-1f8b-11e8-9a40-d4fb26aced3e.png)

**Comment by alexbrainman** [comment]
> this flug "windowsgui" need to cancel the window console

I don't get any console windows. Both at version 1.9 and 1.10.

How did you build your executable? Did you see my command there https://github.com/golang/go/issues/24232#issuecomment-370197648 ?
What is your command? Can I download your program to try it here?

Alex

**Comment by devorel** [comment]
Hi
i try your program , and i get it again
go build -ldflags "-H windowsgui" -v -o %tmp%\a.exe github.com/alexbrainman/gowingui

![v1 10 bug](https://user-images.githubusercontent.com/16879980/36943394-6184bb6e-1f91-11e8-94b7-3d89640e53d3.png)

https://golang.org/dl/
go1.10.windows-amd64.msi | Installer | Windows | x86-64 | 106MB

**Comment by alexbrainman** [comment]
> go build -ldflags "-H windowsgui" -v -o %tmp%\a.exe github.com/alexbrainman/gowingui

Looking at your screenshot, I can see you did something completely different. You used \ instead of / in your import path. Do not do that. Really use
```
go build -ldflags "-H windowsgui" -v -o %tmp%\a.exe github.com/alexbrainman/gowingui
```
command.

Alex

**Comment by devorel** [comment]
i tried but i have the problem yet.
with 1.94 i dont have any problem!!!!

2018-03-04 10:30 GMT+02:00 Alex Brainman <notifications@github.com>:

**Comment by alexbrainman** [comment]
I don't know how to reproduce your problem here. Sorry.

Alex

**Comment by devorel** [comment]
I don't also..
But i say .
You have a problem with the flags in the last version!


בתאריך 4 במרץ 2018 22:51,‏ "Alex Brainman" <notifications@github.com> כתב:

**Comment by asticode** [comment]
I've built a [library](https://github.com/asticode/go-astilectron-bundler) that uses the `-H windowsgui` flag and some of my users are complaining it stopped working in GO 1.10 too. Unfortunately I didn't have time to investigate further but this issue seems to describe an actual problem therefore I'd like to help.

@alexbrainman which windows are you testing on?

**Comment by devorel** [comment]
try download
go1.10.windows-amd64.msi <https://dl.google.com/go/go1.10.windows-amd64.msi>
Installer Windows x86-64 106MB
https://golang.org/dl/

2018-03-06 11:09 GMT+02:00 Alex Brainman <notifications@github.com>:

**Comment by krasnovu** [comment]
@alexbrainman, I have a similar problem on Go 1.10.1 (I could not downgrade to 1.9.3 to check if there was an error on this version) and I`ve found a way to reproduce it.

Steps to reproduce:
1) Check the GOPATH parameter in go env. For example, `E:\Go`.
2) In the console, change the directory using other letter case of part intersecting with GOPATH variable. For example, `cd e:\go\src\site\project` (here "e" and "go" in lowercase).
3) Build the project with the ldflags keys. For example `go build -ldflags "-s -w -H=windowsgui"`.

P.S.: In my auto-builder project the second step looks like `os.Chdir(dir)`.

**Comment by alexbrainman** [comment]
> @alexbrainman, I have a similar problem on Go 1.10.1 (I could not downgrade to 1.9.3 to check if there was an error on this version) and I`ve found a way to reproduce it.

Yes, I can reproduce it here too now. I use these commands:
```
set MYHOME=c:\users\alex\dev
set GOROOT=%MYHOME%\go
set GOROOT_BOOTSTRAP=%MYHOME%\go1.4.3
set GOPATH=%MYHOME%
set PATH=%PATH%;%GOROOT%\bin
cd %GOPATH%\src\github.com\alexbrainman\gowingui
go build -ldflags "-H windowsgui" -v -o %tmp%\a.exe
pause
```
That creates %tmp%\a.exe that opens command window when it runs.

Thank you @krasnovu 

Alex

**Comment by alexbrainman** [comment]
Just submitted the fix. Should we back port the fix to go1.10 and go1.9 ?

Alex

**Comment by ianlancetaylor** [comment]
@alexbrainman We shouldn't backport to 1.9 which doesn't have this problem anyhow.

A backport to 1.10 is OK with me if you think it is a good idea.

**Comment by alexbrainman** [comment]
> We shouldn't backport to 1.9 which doesn't have this problem anyhow.

I agree, go1.9 is not broken with this issue, so we do not need to backport to go1.9.

> A backport to 1.10 is OK with me if you think it is a good idea.

This issue did not affect me personally, so I do not know if it should be back ported or not. That is why I asked here. And it looks like no one cares about backporting this to go1.9. So I assume we don't want to backport this until we hear some complains.

Thank you, Ian.

Alex
